core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -201,10 +201,18 @@ class SnappySession(_sc: SparkContext) extends SparkSession(_sc) {
     }
   }
 
-  final def prepareSQL(sqlText: String): LogicalPlan = {
+  final def prepareSQL(sqlText: String, skipPromote: Boolean = false): LogicalPlan = {
     val logical = sessionState.sqlParser.parsePlan(sqlText, clearExecutionData = true)
     SparkSession.setActiveSession(this)
-    sessionState.analyzerPrepare.execute(logical)
+    var ap: Analyzer = null
+    if (!skipPromote) {
+      ap = sessionState.analyzerPrepare
+    } else {
+      ap = sessionState.analyzerWithoutPromote
+    }
+    // logInfo(s"KN: Batches ${ap.batches.filter(
+    //  _.name.equalsIgnoreCase("Resolution")).mkString("_")}")
+    ap.execute(logical)
```

Comment:
Do we need analyzerPrepare now or can we just use org.apache.spark.sql.internal.SnappySessionState#analyzer?